### DISCUSSION TITLE
AZD Template to Deploy from Container Registry

### DISCUSSION THREAD

**Discussion by kurt-mueller-osumc** [Azure/azure-dev#1114] [Q&A] [answered]
I have a rails app that I package up as a docker image in an Azure container registry. Once that image gets built, app service deploys that image as an app. I'd love to be able to use an AZD template do to this or build my own.

Anybody know where to start?

**Comment by lechnerc77** [comment, marked as answer]
Hi Kurt,

I think your setup is somewhere in between the scenarios provided as templates for deploying an app to the Azure App Service and the deployment to Azure Container Apps (when it comes to the build and the storage in the Azure Container Registry). These templates are certainly worth a look: [https://learn.microsoft.com/en-us/azure/developer/azure-developer-cli/azd-templates?tabs=nodejs#choose-a-template](https://learn.microsoft.com/en-us/azure/developer/azure-developer-cli/azd-templates?tabs=nodejs#choose-a-template)`. However, no template is a perfect fit for your use case, so you will probably need to build your own.

In addition the configuration of a deployment pipeline is something you should look into ([https://learn.microsoft.com/en-us/azure/developer/azure-developer-cli/configure-devops-pipeline?tabs=GitHub](https://learn.microsoft.com/en-us/azure/developer/azure-developer-cli/configure-devops-pipeline?tabs=GitHub))

Cheers Christian
